packages/integration/lib/src/views/poll/handler/poll_handler.dart

```diff
@@ -0,0 +1,366 @@
+                PollMultiSelectState.atMax &&
+            options.length > attachmentMeta.multiSelectNo!) {
+          LMFeedCore.showSnackBar(
+            context,
+            "Please select at most ${attachmentMeta.multiSelectNo} options",
+            source,
+          );
+
+          attachmentMeta = previousValue;
+          resetOptions(attachmentMeta, previousValue);
+          return;
+        }
+      }
+      int totalVotes = attachmentMeta.options?.fold(0,
+              (previousValue, element) => previousValue! + element.voteCount) ??
+          0;
+      for (int i = 0; i < attachmentMeta.options!.length; i++) {
+        if (options.contains(attachmentMeta.options![i].id)) {
+          attachmentMeta.options![i].isSelected = true;
+          attachmentMeta.options![i].voteCount++;
+          totalVotes++;
+          attachmentMeta.options![i].percentage =
+              (attachmentMeta.options![i].voteCount / totalVotes) * 100;
+        } else if (previousValue.options![i].isSelected) {
+          attachmentMeta.options![i].isSelected = false;
+          attachmentMeta.options![i].voteCount--;
+          totalVotes--;
+          attachmentMeta.options![i].percentage =
+              (attachmentMeta.options![i].voteCount / totalVotes) * 100;
+        }
+      }
+      rebuildPollWidget.value = !rebuildPollWidget.value;
+      SubmitPollVoteRequest request = (SubmitPollVoteRequestBuilder()
+            ..pollId(attachmentMeta.id ?? '')
+            ..votes([...options]))
+          .build();
+      final response = await LMFeedCore.client.submitPollVote(request);
+      if (!response.success) {
+        for (int i = 0; i < options.length; i++) {
+          int index = attachmentMeta.options!
+              .indexWhere((element) => element.id == options[i]);
+          if (index != -1) {
+            attachmentMeta.options![index].isSelected = false;
+            attachmentMeta.options![index].voteCount--;
+            attachmentMeta.options![index].percentage =
+                (attachmentMeta.options![index].voteCount / totalVotes) * 100;
+          }
+        }
+
+        attachmentMeta = previousValue;
+        resetOptions(attachmentMeta, previousValue);
+        LMFeedCore.showSnackBar(
+          context,
+          response.errorMessage ?? "",
+          source,
+        );
+      } else {
+        LMFeedCore.showSnackBar(
+          context,
+          "Vote submitted successfully",
+          source,
+        );
+        PostDetailRequest postDetailRequest = (PostDetailRequestBuilder()
+              ..postId(postId)
+              ..pageSize(10)
+              ..page(1))
+            .build();
+        final postResponse =
+            await LMFeedCore.client.getPostDetails(postDetailRequest);
+        final Map<String, LMUserViewData> users = postResponse.users?.map(
+                (key, value) =>
+                    MapEntry(key, LMUserViewDataConvertor.fromUser(value))) ??
```

Comment:
widgets data should be passed in topic view data convertor

Reply:
done